app/src/common/shared/com/igalia/wolvic/downloads/DownloadsManager.java

```diff
@@ -104,15 +105,21 @@ public void startDownload(@NonNull DownloadJob job) {
 
         Uri url = Uri.parse(job.getUri());
         DownloadManager.Request request = new DownloadManager.Request(url);
+
         request.setTitle(job.getTitle());
         request.setDescription(job.getDescription());
         request.setMimeType(job.getContentType());
         request.setNotificationVisibility(DownloadManager.Request.VISIBILITY_VISIBLE_NOTIFY_COMPLETED);
         request.setVisibleInDownloadsUi(false);
 
-        if (job.getOutputPath() == null) {
+        if (job.getOutputPath()==null) {
             try {
-                request.setDestinationInExternalFilesDir(mContext, Environment.DIRECTORY_DOWNLOADS, job.getFilename());
+                //Changed the line to PublicDir, and added some permissions.
+                //the file gets stored in Downloads, but, we cannot open it from within wolvic.
+                request.setDestinationInExternalPublicDir(Environment.DIRECTORY_DOWNLOADS, job.getFilename());
```

Comment:
These comments are fine for local changes while developing but we should not include them in the final patch, so please remove them.

Reply:
> These comments are fine for local changes while developing but we should not include them in the final patch, so please remove them.

Will keep this in mind thanks for mentioning it :)